lib/asn1/der_copy.c

```diff
@@ -103,7 +103,9 @@ der_copy_printable_string (const heim_printable_string *from,
     to->data   = malloc(to->length + 1);
     if(to->data == NULL)
 	return ENOMEM;
-    memcpy(to->data, from->data, to->length);
+    if (from->data != NULL) {
```

Comment:
This should be on the callers, not here.

Reply:
Either the callers should check or we should have an `assert()` here about both, `from` and `to` (`assert(from && to);`).  I would prefer the assertion.